Fixes and update to the dataset and dataset status entry points

Amli opened the pull request:
- use page parameter instead of start for datasets list
- add the get dataset entry point
- remove duplicated dataset_status attribute in dataset object description
- add a dataset status object example
- add a notice for about dataset status API that will be modified

Amli committed:
Fixes and update to the dataset and dataset status entry points

- use page parameter instead of start for datasets list
- add the get dataset entry point
- remove duplicated dataset_status attribute in dataset object description
- add a dataset status object example
- add a notice for about dataset status API that will be modified

jmorel commented on source/includes/_dataset_status.md:
```diff
@@ -2,6 +2,25 @@
+
+<aside class="notice">
+This API needs to be refactored into something clearer.
+</aside>
```
I think this should not be in the documentation but rather on a todo list somewhere :)

Amli replied:
I thought it was a good idea to let people know in advance that we are going to break this entry point in the future
but no worries, I'll remove the notice

jmorel commented on source/includes/_dataset_status.md:
```diff
@@ -19,7 +38,7 @@ Attribute | Description
 `name` <br> *string* | One of the dataset status values
 `since` <br> *datetime* | Timestamp when the dataset entered in the current status
-`user` <br> *user* <br> <em class="expandable">expandable</em> | User who started the action
+`user` <br> *user* <br> <em class="expandable">expandable</em> | User who started the action (currently not available in the API)
```
Let's remove it entirely then, I really think the documentation should reflect the current state of the API, not a desired state.

Also, we need to add previous and next in this list since they are available.
And we need to add previous and next attributes to all examples in this section.

juliendbg replied:
As I understood it, "next" is not implemented yet, so let's not consider it in the current state of the API neither.

jmorel commented on source/includes/_datasets.md:
```diff
@@ -41,7 +41,7 @@ Through the management API, it is possible to:
 
 - the `dataset_uid` that is automatically set, and will never change through the lifetime of the dataset.
-- the `dataset_id`, that can be chosen during dataset creation or changed on an unpublished dataset
+- the `dataset_id`, that can be chosen during dataset creation or changed on an unpublished dataset. It will be used in the exploration API and UI to access the dataset with a meaningful URL.
```
s/exploration/explore
I don't think I've seen "exploration" anywhere, we always say "explore" (that may be a bad practice but coherence etc.)

Amli replied:
👍

jmorel commented on source/includes/_datasets.md:
```diff
@@ -105,12 +104,52 @@ This endpoint lists all the datasets that can be edited by this user.
+            "language": "en",
+            "title": "My Dataset",
+        }
+    },
```
Isn't metas an expandable property ? If so the URL should reflect that with an "expand" parameter.

Amli replied:
true

juliendbg commented on source/includes/_datasets.md:
```diff
@@ -105,12 +104,52 @@ This endpoint lists all the datasets that can be edited by this user.
 --------- | ------- | -----------
 `where` <br> *string* | None | Filter expression used to restrict returned datasets ([ODSQL documentation](https://docs.opendatasoft.com/api/explore/v2.html#where-clause))
-`start` <br> *string* | 0 | Index of the first item to return
+`page` <br> *string* | 1 | Number of the page you want to retrieve. Pages numbers vary when changing the number of rows to retrieve.
```
The 2nd part of the description confused me, I had to reread it several times to understand it. I think we can consider people know what a page is, no?

Amli replied:
No problem, I'll drop it

juliendbg commented on source/includes/_datasets.md:
```diff
@@ -105,12 +104,52 @@ This endpoint lists all the datasets that can be edited by this user.
+Returns a list of dataset objects.
+
+
+## Retrieve information about one dataset
```
In the explore documentation, we use "Looking up a dataset", maybe we can use the same vocabulary in both?

jmorel replied:
I agree we should be using the same vocabulary, but in this case I'd go for retrieve instead of look up. If I'm not mistaken, "look up" means "search" and retrieve is closer to "get".

Amli replied:
agreed, I would prefer we change the wording in the explore documentation

Amli committed:
Dataset and dataset status cleanup

- remove unclear attribute description for page
- add previous/next attributes to dataset status examples
- add ?expand=metas to dataset retrieve example
- specify the metadata parameter for dataset is expandable
- s/exploration/explore
- remove 'dataset status API needs refactor' notice

jmorel commented:
I see you've added the "next" property to all status example. I thought that since it wasn't properly implemented we had to remove it (cf @juliendbg message in an earlier review). Otherwise 👍

Amli commented:
It is implemented, you actually get the `next` property when asking for status. It's just probably not implemented everywhere

meganelacheny commented on source/includes/_datasets.md:
```diff
@@ -105,12 +96,52 @@ This endpoint lists all the datasets that can be edited by this user.
 --------- | ------- | -----------
 `where` <br> *string* | None | Filter expression used to restrict returned datasets ([ODSQL documentation](https://docs.opendatasoft.com/api/explore/v2.html#where-clause))
-`start` <br> *string* | 0 | Index of the first item to return
+`page` <br> *string* | 1 | Number of the page you want to retrieve.
```
Try to avoid using "you" in the documentation. "Number of the page to retrieve" would be better.

Amli commented:
outdated
